File: projects/breathflow/Models/BreathingTechnique.swift
// Models/BreathingTechnique.swift
import Foundation

enum BreathingTechnique: String, CaseIterable, Codable, Sendable {
    case fourSevenEight
    case boxBreathing
    case calmBreathing
    
    var displayName: String {
        switch self {
        case .fourSevenEight:
            return "4-7-8 Breathing"
        case .boxBreathing:
            return "Box Breathing"
        case .calmBreathing:
            return "Calm Breathing"
        }
    }
    
    var description: String {
        switch self {
        case .fourSevenEight:
            return "Deep relaxation technique with extended hold phase"
        case .boxBreathing:
            return "Balanced breathing pattern used by Navy SEALs"
        case .calmBreathing:
            return "Gentle pattern for everyday stress relief"
        }
    }
    
    var icon: String {
        switch self {
        case .fourSevenEight:
            return "lungs.fill"
        case .boxBreathing:
            return "square.fill"
        case .calmBreathing:
            return "leaf.fill"
        }
    }
    
    var inhaleSeconds: Int {
        switch self {
        case .fourSevenEight:
            return 4
        case .boxBreathing:
            return 4
        case .calmBreathing:
            return 5
        }
    }
    
    var holdSeconds: Int {
        switch self {
        case .fourSevenEight:
            return 7
        case .boxBreathing:
            return 4
        case .calmBreathing:
            return 5
        }
    }
    
    var exhaleSeconds: Int {
        switch self {
        case .fourSevenEight:
            return 8
        case .boxBreathing:
            return 4
        case .calmBreathing:
            return 5
        }
    }
    
    var totalCycleDuration: Int {
        inhaleSeconds + holdSeconds + exhaleSeconds
    }
}

File: projects/breathflow3_android/Models/BreathingTechnique.kt
package com.driveai.breathflow.data.models

enum class BreathingTechnique(
    val displayName: String,
    val description: String,
    val inhaleDurationMs: Int,
    val holdDurationMs: Int,
    val exhaleDurationMs: Int,
    val totalCycleDurationMs: Int
) {
    FOUR_SEVEN_EIGHT(
        displayName = "4-7-8 Breathing",
        description = "Slow, deep breath for calming anxiety",
        inhaleDurationMs = 4000,
        holdDurationMs = 7000,
        exhaleDurationMs = 8000,
        totalCycleDurationMs = 19000
    ),
    BOX_BREATHING(
        displayName = "Box Breathing",
        description = "Equal counts for balance and focus",
        inhaleDurationMs = 4000,
        holdDurationMs = 4000,
        exhaleDurationMs = 4000,
        totalCycleDurationMs = 12000
    ),
    CALM_BREATHING(
        displayName = "Calm Breathing",
        description = "Gentle, natural rhythm for relaxation",
        inhaleDurationMs = 3000,
        holdDurationMs = 2000,
        exhaleDurationMs = 4000,
        totalCycleDurationMs = 9000
    );

    fun getDurationInSeconds(): Int = totalCycleDurationMs / 1000
}

enum class BreathingPhase {
    IDLE,
    INHALE,
    HOLD,
    EXHALE,
    COMPLETE
}

data class SessionRecord(
    val id: String = "",
    val technique: String = "",
    val durationSeconds: Int = 0,
    val cyclesCompleted: Int = 0,
    val timestamp: Long = System.currentTimeMillis(),
    val notes: String = ""
)